combat-trainer.lic

```diff
@@ -391,9 +394,19 @@ class LootProcess
     end
   end
 
-  def should_perform_ritual(prepare_consume)
-    !prepare_consume &&
-      ((@necro_train_first_aid && DRSkill.getxp('First Aid') < 32) || DRSkill.getxp('Thanatology') < 32)
+  def can_perform_training_ritual?(game_state)
+    return false unless can_perform_priority_ritual?(game_state)
+    return false if @necro_train_first_aid && DRSkill.getxp('First Aid') < 32
+    return false if DRSkill.getxp('Thanatology') < 32
+    true
+  end
+
+  def can_perform_priority_ritual?(game_state)
```

Comment:
Priority rituals are healing and zombie making rituals. They are 'priority' because they do something besides training

Reply:
Is this similar to `game_state.necro_casting?`?

Reply:
Hmm yea theyre pretty much the same. I will remove `necro_casting`

Reply:
This feels a bit like [feature envy](https://softwareengineering.stackexchange.com/questions/212128/what-is-a-feature-envy-code-and-why-is-it-considered-a-code-smell) to me. What do you think?

Reply:
I see it. I think it might be worth moving this check to game_state